### ISSUE TITLE
Fatal Error when running Terraform Plan in Azure DevOps Release Pipeline

### ISSUE THREAD

**Issue by dylanx77** [hashicorp/terraform#23677]
### Terraform Version 

```
0.12.18
```

### Terraform Configuration Files

```
network.tf - main config file

terraform {
  backend "gcs" {
      bucket = var.TerraformBackendBucket
      prefix = var.TerraformBackendBucketPrefix
  }
}

# Create Network
resource "google_compute_network" "private_network" {
  name                    = var.PrivateNetworkName
  auto_create_subnetworks = "false"
  routing_mode            = var.PrivateNetworkRoutingMode
}

# Create us-east1-b subnetwork
resource "google_compute_subnetwork" "us-east1-b-sn" {
  name          = var.USEast1BSnName
  region        = var.USEast1BSnRegion
  network       = var.PrivateNetworkSelfLink
  ip_cidr_range = var.USEast1BSnRegion
}

# Create us-east1-c subnetwork
resource "google_compute_subnetwork" "us-east1-c-sn" {
  name          = var.USEast1CSnName
  region        = var.USEast1CSnRegion
  network       = var.PrivateNetworkSelfLink
  ip_cidr_range = var.USEast1CSnCIDR
}

# Create eu-west1-a subnetwork
resource "google_compute_subnetwork" "eu-west1-b-sn" {
  name          = var.USEast1CSnName
  region        = var.USEast1CSnRegion
  network       = var.PrivateNetworkSelfLink
  ip_cidr_range = var.EuWest1BSnCIDR
}
```

```
terraform.tfvars.json - variable initialization file

{
"Project": "__Gcp.Project__",
"Region": "__Gcp.Region__",
"Zone": "__Gcp.Zone__",
"TerraformVersion": "__Terraform.Version__",
"TerraformProvider": "__Terraform.Provider__",
"TerraformCommand": "__Terraform.Command__",
"ConfigurationDirectory": "__Configuration.Directory__",
"ServiceConnection": "__Gcp.Service.Connection__",
"TerraformBackendBucket": "__Terraform.Backend.Bucket__",
"TerraformBackendBucketPrefix": "__Terraform.Backend.Bucket.Prefix__",
"PrivateNetworkName": "__Private.Network.Name__",
"PrivateNetworkSelfLink": "__Private.Network.SelfLink__",
"PrivateNetworkRoutingMode": "__Private.Network.Routing.Mode__",
"USEast1BSnName": "__UsEast1B.Sn.Name__",
"USEast1BSnRegion": "__UsEast1B.Sn.Region__",
"USEast1BSnCIDR": "__UsEast1B.Sn.CIDR__",
"USEast1CSnName": "__UsEast1C.Sn.Name__",
"USEast1CSnRegion": "__UsEast1C.Sn.Region__",
"USEast1CSnCIDR": "__UsEast1C.Sn.CIDR__",
"EuWest1BSnName": "__EuWest1B.Sn.Name__",
"EuWest1BSnRegion": "__EuWest1B.Sn.Region__",
"EuWest1BSnCIDR": "__EuWest1B.Sn.CIDR__"
}
```

```
variables.tf - variable declaration file

variable "Project" {
  type        = string
  description = "Gcp project to deploy to"
}

variable "Region" {
  type        = string
  description = "Gcp region to deploy to"
}

variable "Zone" {
  type        = string
  description = "Gcp zone to deploy to"
}

variable "TerraformVersion"{
  type        = string
  description = "Terraform version"
}

variable "TerraformProvider"{
  type        = string
  description = "Terraform provider - Aws, Gcp, Azure"
}

variable "TerraformCommand"{
  type        = string
  description = "Terraform command to execute - init, plan, apply"
}

variable "ConfigurationDirectory"{
  type        = string
  description = "Current working directory"
}

variable "ServiceConnection"{
  type        = string
  description = "Provider service connection"
}

variable "TerraformBackendBucket" {
  type        = string
  description = "Gcp bucket for state file storage"
}

variable "TerraformBackendBucketPrefix" {
  type        = string
  description = "Gcp bucket prefix (bucketname/prefix)"
}

variable "PrivateNetworkName" {
  type        = string
  description = "Gcp VPC name"
}

variable "PrivateNetworkSelfLink" {
  type        = string
  description = ""
}

variable "PrivateNetworkRoutingMode"{
  type        = string
  description = "Routing mode for private network - GLOBAL or REGIONAL"
}

variable "USEast1BSnName"{
  type        = string
  description = ""
}

variable "USEast1BSnRegion"{
  type        = string
  description = ""
}

variable "USEast1BSnCIDR"{
  type        = list(string)
  description = ""
}

variable "USEast1CSnName"{
  type        = string
  description = ""
}

variable "USEast1CSnRegion"{
  type        = string
  description = ""
}

variable "USEast1CSnCIDR"{
  type        = list(string)
  description = ""
}

variable "EuWest1BSnName"{
  type        = string
  description = ""
}

variable "EuWest1BSnRegion"{
  type        = string
  description = ""
}

variable "EuWest1BSnCIDR"{
  type        = list(string)
  description = ""
}
```

### Debug Output
Running Terraform in an Azure DevOps pipeline. Unsure how to obtain debug info.

### Crash Output
2019-12-14T13:35:40.6293306Z ##[section]Starting: Terraform : gcp
2019-12-14T13:35:40.6397960Z ==============================================================================
2019-12-14T13:35:40.6398066Z Task         : Terraform
2019-12-14T13:35:40.6398133Z Description  : Execute terraform commands to manage resources on AzureRM, Amazon Web Services(AWS) and Google Cloud Platform(GCP)
2019-12-14T13:35:40.6398212Z Version      : 0.0.137
2019-12-14T13:35:40.6398270Z Author       : Microsoft Corporation
2019-12-14T13:35:40.6398347Z Help         : [Learn more about this task](https://aka.ms/AA5j5pf)
2019-12-14T13:35:40.6399012Z ==============================================================================
2019-12-14T13:35:40.8777513Z [command]C:\hostedtoolcache\windows\terraform\0.12.18\x64\terraform.exe providers
2019-12-14T13:35:41.3849575Z .
2019-12-14T13:35:41.3861997Z └── provider.google
2019-12-14T13:35:41.3862126Z 
2019-12-14T13:35:41.3912243Z [command]C:\hostedtoolcache\windows\terraform\0.12.18\x64\terraform.exe plan
2019-12-14T13:35:41.5263004Z panic: runtime error: index out of range
2019-12-14T13:35:41.5263317Z 
2019-12-14T13:35:41.5263845Z goroutine 1 [running]:
2019-12-14T13:35:41.5310108Z github.com/hashicorp/hcl/v2/json.(*peeker).Read(...)
2019-12-14T13:35:41.5310401Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl/v2@v2.0.0/json/peeker.go:20
2019-12-14T13:35:41.5310610Z github.com/hashicorp/hcl/v2/json.parseObject.func1(0x0, 0xc00037a1fa, 0x1, 0x2ec, 0x1edad7c, 0x15, 0x11, 0x21, 0x1fa, 0x11, ...)
2019-12-14T13:35:41.5310831Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl/v2@v2.0.0/json/parser.go:127 +0xb0
2019-12-14T13:35:41.5311021Z github.com/hashicorp/hcl/v2/json.parseObject(0xc000135900, 0xc000127920, 0x1edad7c, 0x15, 0x30000, 0x13)
2019-12-14T13:35:41.5311202Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl/v2@v2.0.0/json/parser.go:233 +0xcb9
2019-12-14T13:35:41.5311376Z github.com/hashicorp/hcl/v2/json.parseValue(0xc000135900, 0xc000135900, 0x2ec, 0x1edad7c, 0x15, 0x11)
2019-12-14T13:35:41.5311711Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl/v2@v2.0.0/json/parser.go:45 +0x712
2019-12-14T13:35:41.5311891Z github.com/hashicorp/hcl/v2/json.parseFileContent(0xc00037a000, 0x2e6, 0x4e6, 0x1edad7c, 0x15, 0x0, 0x0, 0x43f550, 0x2030000, 0x2030000)
2019-12-14T13:35:41.5312090Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl/v2@v2.0.0/json/parser.go:21 +0x15c
2019-12-14T13:35:41.5312272Z github.com/hashicorp/hcl/v2/json.Parse(0xc00037a000, 0x2e6, 0x4e6, 0x1edad7c, 0x15, 0x0, 0x0, 0xc0001358c0, 0x7ff98e15f070)
2019-12-14T13:35:41.5312456Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl/v2@v2.0.0/json/public.go:21 +0x7e
2019-12-14T13:35:41.5312643Z github.com/hashicorp/terraform/command.(*Meta).addVarsFromFile(0xc000292ea0, 0x1edad7c, 0x15, 0xc000000046, 0xc000249410, 0x0, 0x1, 0x408bc7)
2019-12-14T13:35:41.5312812Z 	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/meta_vars.go:158 +0x58f
2019-12-14T13:35:41.5313006Z github.com/hashicorp/terraform/command.(*Meta).collectVariableValues(0xc000292ea0, 0xc0000b2870, 0x0, 0x0, 0xc00012b520)
2019-12-14T13:35:41.5313187Z 	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/meta_vars.go:70 +0xadb
2019-12-14T13:35:41.5313353Z github.com/hashicorp/terraform/command.(*PlanCommand).Run(0xc000292ea0, 0xc000074430, 0x0, 0x0, 0xc00006bd70)
2019-12-14T13:35:41.5313540Z 	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/plan.go:111 +0xc96
2019-12-14T13:35:41.5313705Z github.com/mitchellh/cli.(*CLI).Run(0xc000080780, 0xc000080780, 0xc0004f9d90, 0x1)
2019-12-14T13:35:41.5313881Z 	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1f8
2019-12-14T13:35:41.5314030Z main.wrappedMain(0x0)
2019-12-14T13:35:41.5314205Z 	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:238 +0xc3b
2019-12-14T13:35:41.5315471Z main.realMain(0x0)
2019-12-14T13:35:41.5315644Z 	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:102 +0xbb
2019-12-14T13:35:41.5315811Z main.main()
2019-12-14T13:35:41.5315971Z 	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:38 +0x42
2019-12-14T13:35:41.5316096Z 
2019-12-14T13:35:41.5316324Z 
2019-12-14T13:35:41.5316683Z 
**2019-12-14T13:35:41.5316939Z !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
2019-12-14T13:35:41.5317268Z 
2019-12-14T13:35:41.5317426Z Terraform crashed! This is always indicative of a bug within Terraform.
2019-12-14T13:35:41.5317619Z A crash log has been placed at "crash.log" relative to your current
2019-12-14T13:35:41.5317794Z working directory. It would be immensely helpful if you could please
2019-12-14T13:35:41.5317949Z report the crash with Terraform[1] so that we can fix this.
2019-12-14T13:35:41.5318082Z 
2019-12-14T13:35:41.5318248Z When reporting bugs, please include your terraform version. That
2019-12-14T13:35:41.5318417Z information is available on the first line of crash.log. You can also
2019-12-14T13:35:41.5318571Z get it by running 'terraform --version' on the command line.
2019-12-14T13:35:41.5318691Z 
2019-12-14T13:35:41.5318855Z SECURITY WARNING: the "crash.log" file that was created may contain 
2019-12-14T13:35:41.5319009Z sensitive information that must be redacted before it is safe to share 
2019-12-14T13:35:41.5319182Z on the issue tracker.
2019-12-14T13:35:41.5319300Z 
2019-12-14T13:35:41.5319461Z [1]: https://github.com/hashicorp/terraform/issues
2019-12-14T13:35:41.5319581Z 
2019-12-14T13:35:41.5319743Z !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!**
2019-12-14T13:35:41.5389044Z ##[error]Error: The process 'C:\hostedtoolcache\windows\terraform\0.12.18\x64\terraform.exe' failed with exit code 2
2019-12-14T13:35:41.5443708Z ##[section]Finishing: Terraform : gcp

### Expected Behavior
Successful plan and deployment. 

### Actual Behavior
Fatal Crash

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:

![Screen Shot 2019-12-14 at 8 50 40 AM](https://user-images.githubusercontent.com/45051156/70849609-d5b65480-1e4e-11ea-9516-2e66e4948c66.png)

The above screen shot depicts the release pipeline upon which this error occurs. The error is occuring at the Terraform Plan step. (Terraform Init succeeds, Terraform Validate and Apply does not get executed.) There is an error on the replace variables step, but I have had a similar error previously and the deployment was successful. 

### Additional Context
Terraform is running in Azure DevOps as previously stated. I am using a variable group to pass in variables at runtime. That is what occurs in the replace variables step of the release pipeline. With that said, variables are being replaced correctly per the logs in Azure DevOps.

**Comment by danieldreier** [comment]
@dylanx77 I was triaging our crashes and noticed that this seems extremely similar to #24824. Since setting up an Azure DevOps release pipeline and GCP to reproduce this is quite a bit of setup, can you please first confirm that this happens consistently with this terraform config? Does it happen with other terraform configs? It is intermittent?

**Comment by apparentlymart** [comment]
Hi @dylanx77,

We didn't hear back from you after @danieldreier's recent comment. I agree with him that this looks very similar to what's being discussed in #24824, so I'm going to close this one to consolidate the discussion over there. As I mentioned in a comment on the other issue, we have a fix in the main branch to be included in the forthcoming Terraform v0.13.0 which I suspect will fix the problem, but we've not yet proven that. If you're able to help verify that and leave some notes in the other issue to describe what you learned, that'd be much appreciated. Thanks!
